### PR TITLE
Hooks

### LINKED ISSUE #101: Write custom hook
Write custom hook to add and remove sprout cards from local storage

### PR DESCRIPTION
**Pull request by MagdaBer**
- write useLocalStorage hook
- write custom hook to add and remove sprout cards
- write types for sprouts

-closes #101

### CODE REVIEW

File: src/app/hooks/useSproutCards.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  addCard: (sprout: Sprout) => void;
+  removeCard: (newSprout: Sprout) => void;
+} {
+  const [sprouts, setSprouts] = useLocalStorage<Sprout[]>('sprouts', []);
+  function addCard(sprout: Sprout) {
+    setSprouts([...sprouts, sprout]);
+  }
+  function removeCard(newSprout: Sprout) {
+    setSprouts(sprouts.filter((sprout) => sprout !== newSprout));
+  }
```
Comment (marwinburesch) [inline src/app/hooks/useSproutCards.ts:15]: Just a few empty lines after function declarations will look amazing

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export type Sprouts = {
+  sprouts: Sprout[];
+};
+
+export type Sprout = {
+  id: string;
```
Comment (marwinburesch) [inline src/types.ts:6]: could this be a number?

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  image: string;
+  header: string;
+  hours: string;
```
Comment (marwinburesch) [inline src/types.ts:9]: and this too?

File: src/types.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  days: string;
+  textfield: string;
+  ingredients: string;
```
Comment (marwinburesch) [inline src/types.ts:12]: sounds like this could be an array of something

**Review by marwinburesch — COMMENTED** [review]
awesome

**Comment by MagdaBer** [comment]
please check line 10 at Info.tsx. UseParams returns strings and I need a number for the filter in line 12. Thank you

File: src/app/components/CardInfo/CardInfo.stories.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ export default {
 
 export const CardSmall = (): JSX.Element => (
   <CardInfo
-    id="3"
+    id={1}
```
Comment (marwinburesch) [inline src/app/components/CardInfo/CardInfo.stories.tsx:11]: you don't need curlies if it's just a number

File: src/app/pages/Info/Info.tsx [hunk 6]
```diff
@@ -7,8 +7,9 @@ import style from './Info.module.css';
 
 export default function Info(): JSX.Element {
   const { id }: { id: string } = useParams();
+  const idNbr = parseInt(id, 10);
```
Comment (marwinburesch) [inline src/app/pages/Info/Info.tsx:10]: whut's dis? Abbreviations are difficult to read

Comment (MagdaBer) [reply src/app/pages/Info/Info.tsx:10]: please check line 10 at Info.tsx. UseParams returns strings and I need a number for the filter in line 12. Thank you

File: src/app/components/CardInfo/CardInfo.stories.tsx (hunk 5 above)
Comment (MagdaBer) [reply src/app/components/CardInfo/CardInfo.stories.tsx:11]: didn´t work without the curlies

File: src/app/pages/Info/Info.tsx (hunk 6 above)
Comment (phlgr) [inline src/app/pages/Info/Info.tsx:10]:
```suggestion
```

File: src/app/pages/Info/Info.tsx [hunk 7]
```diff
@@ -7,8 +7,9 @@ import style from './Info.module.css';
...
 
-  const filteredSprout = SPROUTS.filter((sprout) => sprout.id === id);
+  const filteredSprout = SPROUTS.filter((sprout) => sprout.id == idNbr);
```
Comment (phlgr) [inline src/app/pages/Info/Info.tsx:12]:
```suggestion
  const filteredSprout = SPROUTS.filter((sprout) => sprout.id === Number(id));
```

**Review by phlgr — CHANGES_REQUESTED** [review]

**Pull request merged**